app/views/projects/blob/_editor.html.haml

```diff
@@ -17,7 +17,10 @@
 
   .file-content.code
     %pre.js-edit-mode-pane#editor
-      = params[:content] || local_assigns[:blob_data]
+      - content = params[:content] || local_assigns[:blob_data]
+      = content
+      - if content.length && 0 === content[content.length-1].index(/[\r\n]/)
```

Comment:
I did some cursory debugging and found the right number of newlines get sent from the UI editor to the controller. Is the issue that the view is truncating the last newline if it's there?